common/src/main/java/com/revenuecat/purchases/common/Dispatcher.kt

```diff
@@ -45,12 +45,24 @@ open class Dispatcher(
     ) {
         synchronized(this.executorService) {
             if (!executorService.isShutdown) {
-                if (useRandomDelay && executorService is ScheduledExecutorService) {
+                val future = if (useRandomDelay && executorService is ScheduledExecutorService) {
                     val delayToApply = (0..JITTERING_DELAY_MILLISECONDS).random()
                     executorService.schedule(command, delayToApply.toLong(), TimeUnit.MILLISECONDS)
                 } else {
-                    executorService.execute(command)
+                    executorService.submit(command)
                 }
+
+                // Exceptions are being swallowed if using execute instead of submit
```

Comment:
maybe we can add a link to the blogpost or docs here?

Reply:
good idea

Reply:
Maybe a link to this PR? Since it describes all the reasoning behind it. What do you think @aboedo ?

Reply:
yeah, that might be better, since we're not actually following the techniques from the blog post